google-cloud-pubsub/lib/google/cloud/pubsub/publish_result.rb

```diff
@@ -0,0 +1,89 @@
+# Copyright 2017 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+module Google
+  module Cloud
+    module Pubsub
+      ##
+      # # PublishResult
+      #
+      class PublishResult
```

Comment:
That is kind of my concern. After all, it also "protects" users from _additions_, which on other APIs has been an issue.

That said, Pub/Sub is super stable by comparison and it is hard to see things being added here, so no need to block on this. I am not sure it is a wise pattern (in other languages it has caused more problems than it solved in my assessment), but I agree that it is widely used in our libraries and this is not a problem case.